### PR TITLE
Support javascript expressions in raw api Selectors (closes #1466)

### LINKED ISSUE #1466: Support javascript expressions in raw api Selectors

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**

### CODE REVIEW

File: src/test-run/execute-js-expression.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { createContext, runInContext } from 'vm';
+import SelectorBuilder from '../client-functions/selectors/selector-builder';
+
+export function executeSelectorExpression (expression, skipVisibilityCheck) {
+    var sandbox = {
+        Selector: (fn, options) => {
```
Comment (churkin) [inline src/test-run/execute-js-expression.js]: `Selector: (fn, options = {})`

File: src/test-run/commands/actions.js [hunk 2]
```diff
@@ -27,6 +30,9 @@ function initSelector (name, val, skipVisibilityCheck) {
         return val;
 
     try {
+        if (val !== null && typeof val === 'object' && val.type === RAW_API_JS_EXPRESSION_TYPE && typeof val.value === 'string')
```
Comment (helen-dikareva) [inline src/test-run/commands/actions.js]: May be better to export this long condition in a var

**Comment by AlexanderMoskovkin** [comment]
FPR

File: src/test-run/execute-js-expression.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { createContext, runInContext } from 'vm';
+import SelectorBuilder from '../client-functions/selectors/selector-builder';
+
+export function executeSelectorExpression (expression, skipVisibilityCheck) {
+    var sandbox = {
+        Selector: (fn, options = {}) => {
+            if (skipVisibilityCheck)
+                options.visibilityCheck = false;
+
+            var builder = new SelectorBuilder(fn, options, { instantiation: 'Selector' });
```
Comment (churkin) [inline src/test-run/execute-js-expression.js]: Here https://github.com/DevExpress/testcafe/blob/c7af41765d752e9d88beaef6e033cddb027292ca/src/test-run/commands/actions.js#L30 we catch errors during the building. Do we need to do this?

Comment (AlexanderMoskovkin) [inline src/test-run/execute-js-expression.js]: Nope, we call this `execute` function from the same `try-catch` block: https://github.com/DevExpress/testcafe/pull/1573/files#diff-5998664e6df837cb69bedb1efa14a399R37

**Comment by AlexanderMoskovkin** [comment]
ping @churkin @helen-dikareva @AndreyBelym

**Comment by AlexanderMoskovkin** [comment]
ping @AndreyBelym

**Pull request merged**